File: 7kyu/(7 kyu) Get the Middle Character/(7 kyu) Get the Middle Character.ts
export class Challenge {
  static getMiddle(s: string) {
    const middle = s.length / 2;
    if (s.length % 2) {
      return s.charAt(Math.floor(middle));
    } else {
      return s.slice(middle - 1, middle + 1);
    }
  }
}

File: 7kyu/(7 kyu) Get the Middle Character/(7 kyu) Get the Middle Character.rs
fn get_middle(s: &str) -> &str {
    // #1
    // let even = s.len() % 2 == 0;
    // let middle = s.len() / 2;
    // &s[if even { middle - 1 } else { middle }..middle + 1]
    // #2
    s[(s.len() - 1) / 2..s.len() / 2 + 1]
}
